From e0e3acb6b56bfe429d20264d97499cb6727fc4f1 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sun, 5 Oct 2025 23:29:10 -0500
Subject: [PATCH 343/347] winewayland: Fix DPI scaling.

I was baited into thinking this doesn't work because rocket league just doesn't support this feature LMAO. Running on winex11 shows that this change produces same result. Therefore, it only makes sense that this is here because it is how upstream wine handles fractional scaling (this is of course not how we are doing it)
---
 dlls/winewayland.drv/wayland_surface.c | 3 +--
 dlls/winewayland.drv/window.c          | 2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_surface.c b/dlls/winewayland.drv/wayland_surface.c
index ddc04e42664..82bb08e2443 100644
--- a/dlls/winewayland.drv/wayland_surface.c
+++ b/dlls/winewayland.drv/wayland_surface.c
@@ -190,8 +190,7 @@ static void wp_fractional_scale_handle_scale(void *user_data,
         if ((surface = data->wayland_surface))
         {
             surface->window.fractional_scale = scale / 120.0;
-            surface->window.scale =
-                surface->window.fractional_scale * NtUserGetSystemDpiForProcess(0) / 96.0;
+            surface->window.scale = surface->window.fractional_scale;
             wayland_surface_refresh_contents(surface);
 
             TRACE("Got scale %lf\n", surface->window.fractional_scale);
diff --git a/dlls/winewayland.drv/window.c b/dlls/winewayland.drv/window.c
index 47a6ae09529..e0ce915063a 100644
--- a/dlls/winewayland.drv/window.c
+++ b/dlls/winewayland.drv/window.c
@@ -181,7 +181,7 @@ static void wayland_win_data_get_config(struct wayland_win_data *data,
         window_state |= WAYLAND_SURFACE_CONFIG_STATE_RESIZEABLE;
 
     conf->state = window_state;
-    conf->scale = conf->fractional_scale * NtUserGetSystemDpiForProcess(0) / 96.0;
+    conf->scale = conf->fractional_scale;
     conf->visible = (style & WS_VISIBLE) == WS_VISIBLE;
     conf->managed = data->managed;
 }
-- 
2.51.0

